tests/run-make/core-ffi-typecheck-clang/tests.rs

```diff
@@ -0,0 +1,93 @@
+// tests.rs
+
+use super::*; // `super` will include everything from `smallcore` once glued together
+
+cfg_if! {
+    if #[cfg(all(target_arch = "aarch64", target_abi = "ilp32"))] {
+        // FIXME: long is not long enough on aarch64 ilp32, should be 8, defaulting to 4
+        const XFAIL_C_LONG_SIZE: usize = 4;
+        pub const TEST_C_LONG_SIZE: () = if size_of::<ffi::c_long>() != XFAIL_C_LONG_SIZE {
+            panic!("wrong c_long size test ilp32");
+        };
+    }
+    else {
+        // Default test
+        pub const TEST_C_LONG_SIZE: () = if size_of::<ffi::c_long>() != CLANG_C_LONG_SIZE {
+            panic!("wrong c_long size");
+        };
+    }
+}
+
+cfg_if! {
+    if #[cfg(target_arch = "csky")] {
+        // FIXME: c_char signedness misallignment on csky, should be signed on CLANG
+        const XFAIL_C_CHAR_SIGNED: bool = false;
+        pub const TEST_C_CHAR_UNSIGNED: () = if ffi::c_char::SIGNED ^ XFAIL_C_CHAR_SIGNED {
```

Comment:
I guess this can use `ffi::c_char::SIGNED != XFAIL_C_CHAR_SIGNED` rather than xor (minicore doesn't need `BitXor` then)